spec/OpenLineage.json

```diff
@@ -210,20 +215,21 @@
         "facets": {
           "description": "The facets for this dataset",
           "type": "object",
-          "additionalProperties": {
-            "anyOf": [
-              { "$ref": "#/$defs/DatasetFacet" },
-              { "$ref": "facets/ColumnLineageDatasetFacet.json" },
-              { "$ref": "facets/DatasourceDatasetFacet.json" },
-              { "$ref": "facets/DataQualityAssertionsDatasetFacet.json" },
-              { "$ref": "facets/LifecycleStateChangeDatasetFacet.json" },
-              { "$ref": "facets/OwnershipDatasetFacet.json" },
-              { "$ref": "facets/SchemaDatasetFacet.json" },
-              { "$ref": "facets/StorageDatasetFacet.json" },
-              { "$ref": "facets/SymlinksDatasetFacet.json" },
-              { "$ref": "facets/DatasetVersionDatasetFacet.json" }
-            ]
-          }
+          "anyOf": [
+            {
+              "type": "object",
+              "additionalProperties": { "$ref": "#/$defs/DatasetFacet" }
+            },
+            { "$ref": "facets/ColumnLineageDatasetFacet.json" },
```

Comment:
Doesn't that mean that `facets` can be of type `facets/DatasourceDatasetFacet.json` ?
In fact, we want to specify that facets are allowed to have `additionalProperties` and those properties can be of a specified types like for example `facets/DatasourceDatasetFacet.json`. This is different: `facets are of certain types` vs `facets have object members of certain types`.

@howardyoo  This does not alter Java client, so nothing gets broken but please double check if this is desired specification.

Reply:
> Doesn't that mean that `facets` can be of type `facets/DatasourceDatasetFacet.json` ? In fact, we want to specify that facets are allowed to have `additionalProperties` and those properties can be of a specified types like for example `facets/DatasourceDatasetFacet.json`. This is different: `facets are of certain types` vs `facets have object members of certain types`.
> 
> @howardyoo This does not alter Java client, so nothing gets broken but please double check if this is desired specification.

So, basically, inside the ColumnLineageDatasetFacet.json https://github.com/OpenLineage/OpenLineage/blob/0ab7d741d5cb8da97e7d1b9c838f498770111496/spec/facets/ColumnLineageDatasetFacet.json#L64
```
  "type": "object",
  "properties": {
    "columnLineage": {
      "$ref": "#/$defs/ColumnLineageDatasetFacet"
    }
  }
```
the allowed property name is `columnLineage` and its ref. type is ColumnLineageDatasetFacet. We want to make sure to have this property name as a legal value for the type, and thus, this specification is not to be treated as `additionalProperties` (additional properties should be reserved for any other properties that we may not know its property key yet).